File: solutions/003/003.js
/**
 * Project Euler - Problem Solution 003
 * Copyright (c) Justin McGettigan. All rights reserved.
 * https://github.com/jwmcgettigan/project-euler-solutions
 */

const isComposite = (num, p) => num >= (p*p);
const isPrime = (num, p) => num % p === 0;
const primeFactorization = (number) => {
  let p = 2;
  while(isComposite(number, p)) {
    if(isPrime(number, p)) {
      number = number / p;
    } else p += 1;
  }
  return number;
}

console.log(primeFactorization(600851475143));

File: solutions/003/003.py
"""
Project Euler - Problem Solution 003
Copyright (c) Justin McGettigan. All rights reserved.
https://github.com/jwmcgettigan/project-euler-solutions
"""

def is_composite(num, p): return num >= (p*p)
def is_prime(num, p): return num % p == 0
def prime_factorization(number):
  p = 2
  while(is_composite(number, p)):
    if(is_prime(number, p)):
      number = number / p;
    else: p += 1
  return number

if __name__ == "__main__":
	print(prime_factorization(600851475143))